### PR TITLE
Enc:support multi slice for AVC encoder on SKL/APL/KBL

### LINKED ISSUE #142: Enable multi slice of AVC encoder on SKL/APL/KBL/GLK
The multi slice feature will be enabled

### PR DESCRIPTION
**Pull request by QuPengfei**
this patch enable the multi slice of AVC encoder. The VA_ENC_SLICE_STRUCTURE_ARBITRARY_MACROBLOCKS is added.

Fixes #142

Signed-off-by: Pengfei Qu <Pengfei.Qu@intel.com>

### CODE REVIEW

File: src/gen9_avc_encoder.c [hunk 1]
```diff
@@ -5399,9 +5435,12 @@ gen9_avc_encode_check_parameter(VADriverContextP ctx,
     /*slice check,all the slices use the same slice height except the last slice*/
     avc_state->arbitrary_num_mbs_in_slice = 0;
     for (i = 0; i < avc_state->slice_num; i++) {
-        assert(avc_state->slice_param[i]->num_macroblocks % generic_state->frame_width_in_mbs == 0);
-        avc_state->slice_height = avc_state->slice_param[i]->num_macroblocks / generic_state->frame_width_in_mbs;
-        /*add it later for muli slices map*/
+        if (avc_state->slice_param[i]->num_macroblocks % generic_state->frame_width_in_mbs > 0) {
+            avc_state->arbitrary_num_mbs_in_slice = 1;
+            avc_state->slice_height = 1;
```
Comment (xhaihao) [inline src/gen9_avc_encoder.c]: why is slice_height 1? for this case?

Comment (QuPengfei) [inline src/gen9_avc_encoder.c]: In this case, the kernel does not care this value of slice_height, here i just put 1 as the default value

Comment (xhaihao) [inline src/gen9_avc_encoder.c]: Could you add a comment in case a reader is confused?

Comment (QuPengfei) [inline src/gen9_avc_encoder.c]: sure

**Comment by QuPengfei** [comment]
update the patch.Please help to review.

**Pull request merged**